Quadriga/src/main/java/edu/asu/spring/quadriga/web/projectblog/AddProjectBlogEntryController.java

```diff
@@ -0,0 +1,120 @@
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import edu.asu.spring.quadriga.aspects.annotations.AccessPolicies;
+import edu.asu.spring.quadriga.aspects.annotations.CheckedElementType;
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.domain.impl.projectblog.ProjectBlogEntry;
+import edu.asu.spring.quadriga.domain.projectblog.IProjectBlogEntry;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.IUserManager;
+import edu.asu.spring.quadriga.service.projectblog.IProjectBlogEntryManager;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+/**
+ * This controller is responsible for providing UI to create project blog entry
+ * and save the entry to database table <code>tbl_projectblogentry</code>.
+ *
+ * @author PawanMahalle
+ */
+@PropertySource(value = "classpath:/user.properties")
+@Controller
+public class AddProjectBlogEntryController {
+
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IProjectBlogEntryManager projectBlogEntryManager;
+
+    @Autowired
+    private IRetrieveProjectManager projectManager;
+
+    /**
+     * provides form to add project blog entry details to the authorized user.
+     * Only project owners and admins are allowed to add entry to the blog.
+     * 
+     * @param projectUnixName
+     *            The project Unix name
+     * @param projectId
+     *            The id assigned to project by framework
+     * @param model
+     *            Model instance
+     * @return view to add new project blog entry
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     *             This exception is thrown when user other than project admin
+     *             and owner tries to add project blog entry.
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "sites/{projectId}/addprojectblogentry", method = RequestMethod.GET)
```

Comment:
You don't have the project id here. You have the unix name. Didn't I comment on this the last time?